FLASK/todolist_api/Repository.py

```diff
@@ -0,0 +1,64 @@
+from file_changes import write_changes_to_file, read_tasks_from_file
+
+
+class JSONRepository:
+    """ A repository class that stores tasks in memory. """
+    def __init__(self):
+        """
+        Initializes the InMemoryRepository by reading tasks from a file.
+        If there's an error reading tasks from the file, it raises an error.
+        """
+        try:
+            self.__data = read_tasks_from_file()
+        except Exception as error:
+            print(f"Error loading tasks from file: {error}")
+
+    def get_all(self):
+        return self.__data  # Returns all the data in the repository
+
+    def get_by_id(self, task_id):
+        task_data = self.__data[task_id].copy()  # Create a copy to avoid modifying the original data
+        return task_data  # Returns the data
+
+    def create(self, attr):
+        task_id = self.generate_task_id()  # Generates ID
+        new_task = {task_id: attr}  # Creates the new task
+
+        self.__data[task_id] = new_task  # Adds the new task to the data
```

Comment:
Daca rulez de 2 ori un create la rand, o sa se pastreze toate datele, inclusiv pe cele create la un get_all?

Reply:
Nu asta face? odata creata daca din parser, si introdusa se duce sa o salveze in fisier. Cand creezi a doua data, prima e deja salvata in fisier. Nu inteleg intrebarea bine..